packages/core/src/storage.ts

```diff
@@ -208,6 +208,47 @@ export type FindResult<T extends Doc> = WithLookup<T>[] & {
 // eslint-disable-next-line @typescript-eslint/no-empty-interface
 export interface TxResult {}
 
+/**
+ * @public
+ */
+export interface IndexedDoc {
+  id: Ref<Doc>
+  _class: Ref<Class<Doc>>
+  space: Ref<Space>
+  modifiedOn: Timestamp
+  modifiedBy: Ref<Account>
+  attachedTo?: Ref<Doc>
+  attachedToClass?: Ref<Class<Doc>>
+  [key: string]: any
+}
+
+/**
+ * @public
+ */
+export interface SearchQuery {
+  query: string
+  // aggregateBy?: string
+  classes?: Ref<Class<Doc>>[]
+  spaces?: Ref<Space>[]
+}
+
+/**
+ * @public
+ */
+export interface SearchOptions {
+  limit?: number
+}
+
+/**
+ * @public
+ */
+export interface SearchResult {
+  // suggest?: any
+  // aggregations?: any
+  docs: IndexedDoc[]
```

Comment:
Why we need IndexedDoc here? I supposed we had a clear structure what is visible on client for mentions and spotlight.

Reply:
We do! Thanks, will fix this